### PR TITLE
[SF7] Refactor console commands in Core Bundle

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
|-----------------|-----
| Branch?         | symfony-7
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ services:
         environment:
             MYSQL_ROOT_PASSWORD: mysql
         ports:
-            - ${MYSQL_PORT:-3306}:3306
+            - ${MYSQL_PORT:-3307}:3306
```
Comment (loic425) [inline docker-compose.yml]: I have to revert that.

File: src/Sylius/Bundle/CoreBundle/Console/Command/InstallDatabaseCommand.php [hunk 2]
```diff
@@ -45,21 +58,34 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
-            ->get('sylius.commands_provider.database_setup')
-            ->getCommands($input, $output, $this->getHelper('question'))
+            ->databaseSetupCommandsProvider
+            ->getCommands($input, $output, $questionHelper)
         ;
 
-        $this->runCommands($commands, $output);
+        $commandExecutor = new CommandExecutor($input, $output, $this->getApplication());
+
+        $commandExecutor->runCommands($commands, $output);
```
Comment (loic425) [inline src/Sylius/Bundle/CoreBundle/Console/Command/InstallDatabaseCommand.php]: If you keep the AbstractInstallCommand, I am wondering if we leave the runCommands inside the CommandExecutor.

File: src/Sylius/Bundle/CoreBundle/Console/Command/InstallDatabaseCommand.php [hunk 3]
```diff
@@ -45,21 +58,34 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
         }
-        $this->commandExecutor->runCommand('sylius:install:sample-data', $parameters, $output);
 
-        return 0;
+        $commandExecutor->runCommand('sylius:install:sample-data', $parameters, $output);
+
+        return Command::SUCCESS;
+    }
+
+    private function getEnvironment(): string
```
Comment (loic425) [inline src/Sylius/Bundle/CoreBundle/Console/Command/InstallDatabaseCommand.php]: Not needed if we keep the AbstractInstallCommand usage.

File: src/Sylius/Bundle/CoreBundle/Console/Command/InstallSampleDataCommand.php [hunk 4]
```diff
@@ -13,21 +13,34 @@
...
+    name: 'sylius:install:sample-data',
+    description: 'Install sample data into Sylius.',
+)]
 final class InstallSampleDataCommand extends AbstractInstallCommand
 {
-    protected static $defaultName = 'sylius:install:sample-data';
+    public function __construct(
+        protected readonly EntityManagerInterface $entityManager,
+        protected readonly CommandDirectoryChecker $commandDirectoryChecker,
+        protected readonly bool $publicDirectoryPath,
```
Comment (loic425) [inline src/Sylius/Bundle/CoreBundle/Console/Command/InstallSampleDataCommand.php]: could be "publicDir" to be closer to "kernel.project_dir" and "kernel.cache_dir" naming.

File: src/Sylius/Bundle/CoreBundle/Installer/Executor/CommandExecutor.php [hunk 5]
```diff
@@ -56,6 +57,30 @@ public function runCommand(string $command, array $parameters = [], ?OutputInter
         return $this;
     }
 
+    /**
+     * @param string[] $commands
+     */
+    public function runCommands(array $commands, OutputInterface $output, bool $displayProgress = true): void
```
Comment (loic425) [inline src/Sylius/Bundle/CoreBundle/Installer/Executor/CommandExecutor.php]: I'm wondering if you still use that cause you keep the AbstractInstallCommand and there is already that method over there.

File: src/Sylius/Bundle/CoreBundle/Installer/Executor/CommandExecutor.php [hunk 6]
```diff
@@ -74,4 +99,16 @@ private function getDefaultParameters(): array
 
         return $defaultParameters;
     }
+
+    protected function createProgressBar(OutputInterface $output, int $length = 10): ProgressBar
```
Comment (loic425) [inline src/Sylius/Bundle/CoreBundle/Installer/Executor/CommandExecutor.php]: this class is final, so eather it should be private or it should be removed if we choose to remove the runCommands method.

**Review by GSadee — APPROVED** [review]

**Pull request merged**